### PR TITLE
fix kubelet data dir do not take effect

### PR DESCRIPTION
**Pull request by x893675**
Signed-off-by: x893675 <x893675@icloud.com>

### What type of PR is this?

/kind bug

### Which issue(s) this PR fixes:

Fixes #

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note
None
```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheme/core/v1/k8s/kubeadm_step.go [hunk 1]
```diff
@@ -396,6 +413,54 @@ func (stepper *KubeadmConfig) InstallSteps(nodes []v1.StepNode) ([]v1.Step, erro
 	}, nil
 }
 
+func (stepper *KubeadmConfig) JoinSteps(isControlPlane bool, nodes []v1.StepNode) ([]v1.Step, error) {
+	stepper.IsControlPlane = isControlPlane
+	kubeadmBytes, err := json.Marshal(stepper)
+	if err != nil {
+		return nil, err
+	}
+	if isControlPlane {
```
[the remark this answers is not in the document]
Comment (x893675) [reply pkg/scheme/core/v1/k8s/kubeadm_step.go:422]: I think it's a bit redundant

[the remark this answers is not in the document]
Comment (x893675) [reply pkg/scheme/core/v1/k8s/kubeadm_step.go:422]: The function returns directly, the assignment statement is a bit redundant

**Review by lixd — APPROVED** [review]
/lgtm

**Pull request merged**